.github/workflows/python-integration.yml

```diff
@@ -0,0 +1,93 @@
+name: Python Integration Tests
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+      ref:
+        required: true
+        type: string
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_TERM_COLOR: always
+  CARGO_INCREMENTAL: 0
+  RUSTFLAGS: "-Dwarnings"
+  CI_REGION: ${{ vars.S3_REGION }}
+  CI_BUCKET: ${{ vars.S3_BUCKET }}
+  CI_PREFIX: ${{ vars.S3_PREFIX }}
+
+jobs:
+  integration-test:
+    name: Integration tests
+    runs-on: ubuntu-22.04
+    environment: ${{ inputs.environment }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+          submodules: true
+          persist-credentials: false
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.ACTIONS_IAM_ROLE }}
+          aws-region: ${{ vars.S3_REGION }}
+
+      - name: Set up stable Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Restore Cargo cache
+        id: restore-cargo-cache
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-${{ github.job }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          # Manually install CPU-only version of torch so we're not carrying around giant GPU drivers/kernels
+          python -m pip install torch --extra-index-url https://download.pytorch.org/whl/cpu
+          python -m pip install -e "s3torchconnectorclient[test,e2e]"
+          python -m pip install -e "s3torchconnector[test,e2e]"
+
+      - name: Dataset integration tests
```

Comment:
nit: Connector Integration Tests might be a better name.

Reply:
I copy-pasted this from our last PR. Can we update it in a fast follow PR?